velox/functions/lib/Re2Functions.cpp

```diff
@@ -978,7 +988,24 @@ std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
       // Ensures that pattern has a single contiguous stream of wildcard
       // characters.
       if (wildcardStart != -1) {
-        return std::make_pair(PatternKind::kGeneric, 0);
+        if (singleCharacterWildcardCount) {
```

Comment:
can you add a comment. if we are inside here then we are sure its not the first time we encounter a sequence of % or _

Reply:
Removed this check and simplified the pattern determination logic, thanks for the feedback.